src/orange/BoundingBoxIO.json.cc

```diff
@@ -0,0 +1,120 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/BoundingBoxIO.json.cc
+//---------------------------------------------------------------------------//
+#include "BoundingBoxIO.json.hh"
+
+#include <cmath>
+#include <limits>
+
+#include "corecel/cont/ArrayIO.json.hh"
+#include "corecel/cont/Range.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+//! Replace "max" with "inf" since the latter can't be represented in JSON.
+template<class T>
+void max_to_inf(celeritas::Array<T, 3>* point)
+{
+    constexpr auto max_real = std::numeric_limits<T>::max();
+    constexpr auto inf = std::numeric_limits<T>::infinity();
+
+    for (auto axis : range(celeritas::Axis::size_))
+    {
+        auto ax = to_int(axis);
+        if (std::fabs((*point)[ax]) == max_real)
+        {
+            (*point)[ax] = std::copysign(inf, (*point)[ax]);
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+//! Replace "inf" with "max" since the latter can't be represented in JSON.
+template<class T>
+void inf_to_max(celeritas::Array<T, 3>* point)
+{
+    constexpr auto max_real = std::numeric_limits<T>::max();
+
+    for (auto axis : range(celeritas::Axis::size_))
+    {
+        auto ax = to_int(axis);
+        if (std::isinf((*point)[ax]))
+        {
+            (*point)[ax] = std::copysign(max_real, (*point)[ax]);
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Read a bounding box from a JSON file.
+ *
+ * A bounding box can be either \c null , indicating an infinite or unknown
```

Comment:
I would say that null doesn't _indicate_ an infinite bounding box, but rather it indicates an missing bbox that we treat as infinite.